### PR TITLE
DEV-26175 Remove explicit condition to check if scrolling options are…

### PR DESCRIPTION
**Pull request by abhijeetnarvekar**
Though the official documentation doesn't say [scrollIntoViewOptions](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) are supported by Safari, the latest version of Safari does support them and they work!

The same maybe true for WaterFox browser 😉 

The hack to check [scrollintoview-browser-compatibility](https://stackoverflow.com/questions/46919627/is-it-possible-to-test-for-scrollintoview-browser-compatibility) may not be true for every browser. The second comment in the hack mentions about safari.

I propose to rely on try-catch.

I have kept the method `isScrollIntoViewCenteredAvailable` as it is exported method and maybe used in some integration.

Let me know your feedback.

### CODE REVIEW

File: src/utils/scrolling.ts [hunk 1]
```diff
@@ -86,12 +86,10 @@ export function isScrollIntoViewCenteredAvailable(): boolean {
  * @return indicates if it succeeded
  */
 export function scrollIntoViewCenteredIfPossible(targetEl: HTMLElement): boolean {
-  if (!isScrollIntoViewCenteredAvailable()) {
```
Comment (pcdeshmukh) [inline src/utils/scrolling.ts:89]: You can get rid of it. Integrator who are using it will stay on old version and with update then need to adjust.

**Comment by acromarco** [comment]
@abhijeetnarvekar 
Just in case we have to support older safaris (maybe we don't ?): 
Have we tested, that the try-catch will really execute the catch for older Safaris, that don't support {block: 'center'} ? 
On the other hand, if it don't throw an exception, it will probably just scrollIntoView without centering which might be sufficient.
In this case it would not execute the dubious workaround  in https://github.com/acrolinx/sidebar-sdk-js/blob/26c9cf84befe20f285dcc7ac06e25b1bfb779602/src/utils/scrolling.ts#L50 
which might be a good thing :smile: after all. (Maybe we should just remove this workaround, at least in the foreseeable future.)
Sorry for this assoziative brain dump.
Long story short: From my point of view, this PR is great and can be merged.

File: src/utils/scrolling.ts (hunk 1 above)
Comment (acromarco) [reply src/utils/scrolling.ts:89]: I had exported `isScrollIntoViewCenteredAvailable` only to use it in a test, which tested that the fallback works for older browsers: https://github.com/acrolinx/sidebar-sdk-js/blob/26c9cf84befe20f285dcc7ac06e25b1bfb779602/test/spec/adapters.ts#L801. 
In this case it scrolls to a different position, so the test must assert a different position if it runs in an older browser.
This said:
 * We could move isScrollIntoViewCenteredAvailable into the test or into test-utils.ts
 * We could use scrollIntoViewCenteredIfPossible with a dummy element as replacement for isScrollIntoViewCenteredAvailable in the test.
 * We could just remove the corresponding assertion in the test, assuming that nobody will run it ever again in an old browser.

**Comment by abhijeetnarvekar** [comment]
Moved `isScrollIntoViewCenteredAvailable` to test utils.

We won't be supporting older Safari. Also the workaround for center scrolling doesn't seem to working on Safari(latest).

Let's depend upon the browser to throw exception if it doesn't support `scrollIntoViewOptions`
